### PR TITLE
Feature/host groups

### PR DESCRIPTION
**Pull request by SoorajBharadwaj07**
# Description
Adding CRUD operations for HostGroup resource

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Feature Pull Request

##### RESOURCE OR DATASOURCE NAME

powermax_host_group

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

### CODE REVIEW

File: powermax/validators.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+			req.AttributePath,
+			v.Description(ctx),
+			fmt.Sprintf("current size : %d", len(elems)),
+		)
+	}
+
+}
+
+func validateSet(ctx context.Context, request tfsdk.ValidateAttributeRequest, response *tfsdk.ValidateAttributeResponse) ([]attr.Value, bool) {
+	var l types.Set
```
Comment (shenda1) [inline powermax/validators.go:49]: Please use meaningful name for the variable

**Comment by naveen-patil-dell** [comment]
run e2e

**Comment by SoorajBharadwaj07** [comment]
run e2e

**Comment by SoorajBharadwaj07** [comment]
run e2e

**Review by shenda1 — APPROVED** [review]

**Pull request merged**